src/plugins/solver.rs

```diff
@@ -373,12 +384,34 @@ fn solve_vel(
             }
 
             if body1.rb.is_dynamic() {
-                body1.linear_velocity.0 += p * inv_mass1;
-                body1.angular_velocity.0 += compute_delta_ang_vel(inv_inertia1, r1, p);
+                let lin_vel_1 = p * inv_mass1;
+                let ang_vel_1 = compute_delta_ang_vel(inv_inertia1, r1, p);
+                if lin_vel_1 != Vector::ZERO {
+                    body1.linear_velocity.0 += lin_vel_1;
+                }
+                #[cfg(feature = "2d")]
+                if ang_vel_1 != 0.0 {
+                    body1.angular_velocity.0 += ang_vel_1;
+                }
+                #[cfg(feature = "3d")]
+                if ang_vel_1 != Vector::ZERO {
+                    body1.angular_velocity.0 += ang_vel_1;
+                }
```

Comment:
- Maybe use `delta_lin_vel`/`delta_ang_vel` here since it's what is used earlier
- I'm not sure if this is more or less readable, but the 2D/3D could technically be combined with `AngularVelocity::ZERO.0`
```suggestion
                let delta_lin_vel = p * inv_mass1;
                let delta_ang_vel = compute_delta_ang_vel(inv_inertia1, r1, p);

                if delta_lin_vel != Vector::ZERO {
                    body1.linear_velocity.0 += delta_lin_vel;
                }
                if delta_ang_vel != AngularVelocity::ZERO.0 {
                    body1.angular_velocity.0 += delta_ang_vel;
                }
```

Reply:
ah good yeah i like`AngularVelocity::ZERO.0` better than the cfg flag